wcfsetup/install/files/lib/system/template/plugin/PagesFunctionTemplatePlugin.class.php

```diff
@@ -172,95 +80,12 @@ public function execute($tagArgs, TemplateEngine $tplObj)
-                $linksBeforePage = 0;
-            }
-            $linksAfterPage = $tagArgs['pages'] - ($tagArgs['page'] + 1);
-            if ($linksAfterPage < 0) {
-                $linksAfterPage = 0;
-            }
-            if ($tagArgs['page'] > 1 && $tagArgs['page'] < $tagArgs['pages']) {
-                $maxLinks--;
-            }
-
-            $half = $maxLinks / 2;
-            $left = $right = $tagArgs['page'];
-            if ($left < 1) {
-                $left = 1;
-            }
-            if ($right < 1) {
-                $right = 1;
-            }
-            if ($right > $tagArgs['pages'] - 1) {
-                $right = $tagArgs['pages'] - 1;
-            }
-
-            if ($linksBeforePage >= $half) {
-                $left -= $half;
-            } else {
-                $left -= $linksBeforePage;
-                $right += $half - $linksBeforePage;
-            }
-
-            if ($linksAfterPage >= $half) {
-                $right += $half;
-            } else {
-                $right += $linksAfterPage;
-                $left -= $half - $linksAfterPage;
-            }
-
-            $right = \intval(\ceil($right));
-            $left = \intval(\ceil($left));
-            if ($left < 1) {
-                $left = 1;
-            }
-            if ($right > $tagArgs['pages']) {
-                $right = $tagArgs['pages'];
-            }
-
-            // left ... links
-            if ($left > 1) {
-                if ($left - 1 < 2) {
-                    $html .= $this->makeLink($link, 2, $tagArgs['page'], $tagArgs['pages']);
-                } else {
-                    $html .= '<li class="jumpTo"><a title="' . WCF::getLanguage()->getDynamicVariable('wcf.page.jumpTo') . '" class="jsTooltip">' . StringUtil::HELLIP . '</a></li>' . "\n";
-                }
-            }
-
-            // visible links
-            for ($i = $left + 1; $i < $right; $i++) {
-                $html .= $this->makeLink($link, $i, $tagArgs['page'], $tagArgs['pages']);
-            }
-
-            // right ... links
-            if ($right < $tagArgs['pages']) {
-                if ($tagArgs['pages'] - $right < 2) {
-                    $html .= $this->makeLink($link, $tagArgs['pages'] - 1, $tagArgs['page'], $tagArgs['pages']);
-                } else {
-                    $html .= '<li class="jumpTo"><a title="' . WCF::getLanguage()->getDynamicVariable('wcf.page.jumpTo') . '" class="jsTooltip">' . StringUtil::HELLIP . '</a></li>' . "\n";
-                }
-            }
-
-            // last page
-            $html .= $this->makeLink($link, $tagArgs['pages'], $tagArgs['page'], $tagArgs['pages']);
-
-            // next page
-            $html .= $this->makeNextLink($link, $tagArgs['page'], $tagArgs['pages']);
-
-            // close div and ul
-            $html .= "</ul></nav>\n";
+            $html = \sprintf(
+                '<woltlab-core-pagination page="%d" count="%d" url="%s"></woltlab-core-pagination>',
```

Comment:
I dislike it a little that the element vanishes entirely without JavaScript, without any indication that something is missing. Perhaps the `.textContent` could be some simple “Page 7/8” text? This makes it a little more obvious that something is amiss in case the JS breaks for whatever reason. Of course this would require undeprecating the template plugin or creating another one with a reasonable API.

Reply:
This would add little to no value but at the same time increases the complexity. The JS breaking apart is most likely the symptom of a much greater issue, so this is acceptable. This does not cover a browser where JS is disabled, but at that point you should prepare yourself for a rough ride anyway.

Reply:
I just remembered that `:defined` exists.

```
woltlab-core-pagination:not(:defined)::after {
    content: 'Page ' attr(page) '/' attr(count);
}
```

would possibly work, but I'll leave this decision up to you :smiley: